### DISCUSSION TITLE
Sharing code with customers?

### DISCUSSION THREAD

**Discussion by PointSingularity** [nrwl/nx#7828] [General]
I am in the process of trying to convince my company to use NX for our (future) projects. After I showed them the functionality of NX, they were certainly impressed and were open to the idea.

Adopting it for new projects is definitely a possibility, but I also have some questions relating to our workflow.

We often share our source code with our customers, but with the monorepo structure, all of our apps/libs would be in the same repo.
Is there a way to extract the source code just from certain apps/libs so only that source code could be shared with the customers?

**Comment by asherccohen** [comment]
It depends. When you say "source code" do you expect to share a runnable app?
It is possible with run-commands to copy files and folders into "dist" and then git push that folder or even publish it to npm.

The tricky part is to maintain a structure that is runnable outside.
Let me know if that's what you're looking for, I can help setting it up.

**Reply by PointSingularity** [reply]
@asherccohen Thanks for your input!

What I've managed for now is that I have a generated dist with a package.json, which works pretty well.

What I was thinking exactly when I made this post is having the same process as above, but without the compiled code, just the source files with a filtered package.json on a per-app basis.

This is possible if I copy all the files with run commands, except for getting the filtered package.json.

Or maybe it is?

**Reply by asherccohen** [reply]
Correct. 
"build" command will always produce compiled code.
If you need source code you should "copy".
